File: Enhancement One New/Appointment.py
from datetime import datetime
import Task
import TaskService
import AppointmentService
import Contact
import ContactService


class Appointment:
    def __init__(self, appointment_id, appointment_date, appointment_desc):
        # no longer than 10 characters and not null
        if appointment_id is None or len(appointment_id) > 10:
            raise ValueError("Invalid ID")
        # no previous date and not null
        if appointment_date is None or appointment_date < datetime.now():
            raise ValueError("Invalid Date")
        # no longer than 50 characters and not null
        if appointment_desc is None or len(appointment_desc) > 50:
            raise ValueError("Invalid Description")

        self.appointment_id = appointment_id
        self.appointment_date = appointment_date
        self.appointment_desc = appointment_desc

    # getters
    def get_id(self):
        return self.appointment_id

    def get_appointment_date(self):
        return self.appointment_date

    def get_appointment_desc(self):
        return self.appointment_desc

    # setters
    def set_appointment_date(self, appointment_date):
        if appointment_date is None or appointment_date < datetime.now():
            raise ValueError("Invalid Date")
        self.appointment_date = appointment_date

    def set_appointment_desc(self, appointment_desc):
        if appointment_desc is None or len(appointment_desc) > 50:
            raise ValueError("Invalid Description")
        self.appointment_desc = appointment_desc

File: Enhancement One Original/projectpackage/Appointment.java
package projectpackage;

import java.util.Date;

public class Appointment {
	
	private String appointmentID;
	private Date appointmentDate;
	private String appointmentDesc;
	
	public Appointment(String id, Date date, String description) {
		//no longer than 10 charac and not null
		if (id == null || id.length() > 10) {
			throw new IllegalArgumentException("Invalid ID");
		}
		//no previous date and not null
		if (date == null || date.before(new Date())) {
			throw new IllegalArgumentException("Invalid Date");
		}
		//no longer than 50 charac and not null
		if (description == null || description.length() > 50) {
			throw new IllegalArgumentException("Invalid Description");
		}
		
		appointmentID = id;
		appointmentDate = date;
		appointmentDesc = description;
	}
	
	//getters
	public String GetID() {
		return appointmentID;
	}
	public Date GetAppointmentDate() {
		return appointmentDate;
	}
	public String GetAppointmentDesc() {
		return appointmentDesc;
	}
	
	//setters
	public void SetAppointmentDate(Date date) {
		if (date == null || date.before(new Date())){
			throw new IllegalArgumentException("Invalid Date");
		}
		appointmentDate = date;
	}
	public void SetAppointmentDesc(String description) {
		if (description == null || description.length() > 50) {
			throw new IllegalArgumentException("Invalid Description");
		}
		appointmentDesc = description;
	}

}
